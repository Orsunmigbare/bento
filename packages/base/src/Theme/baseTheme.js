import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
