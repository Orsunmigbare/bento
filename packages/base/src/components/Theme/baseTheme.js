export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
