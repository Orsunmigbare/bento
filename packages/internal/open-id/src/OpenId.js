import PropTypes from "prop-types"
import React from "react"
import { Consumer } from "./context"

class OpenId extends React.Component {
  render() {
    return <Consumer>{props => this.props.children(props)}</Consumer>
  }
}

OpenId.propTypes = {
  children: PropTypes.func
}

export default OpenId
