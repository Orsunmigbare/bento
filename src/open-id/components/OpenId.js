import loglevel from "loglevel"
import React, { createContext, useContext, useEffect } from "react"
import Oidc, { UserManager } from "oidc-client"
import PropTypes from "prop-types"
import { logLevelFromNumber } from "../utils/logLevelFromString"

export const OpenIdContext = createContext({})

export const OpenId = ({ children, config, extraConfig, level, ...props }) => {
  useEffect(() => {
    Oidc.Log.level = Math.min(level, Oidc.Log.DEBUG) // 4 and 5 produce DEBUG log level
    loglevel.setLevel(logLevelFromNumber(level))
  }, [level])

  const userManager = React.useMemo(() => new UserManager(config), [config])

  const fetchToken = async () => {
    if (!userManager) {
      loglevel.error("UserManager not initialized")
      return
    }
    const user = await userManager.getUser()
    return user?.access_token
  }

  return (
    <OpenIdContext.Provider
      value={{
        userManager,
        config,
        extraConfig,
        fetchToken,
        ...props
      }}
    >
      {children}
    </OpenIdContext.Provider>
  )
}

OpenId.propTypes = {
  level: PropTypes.oneOf([
    Oidc.Log.NONE,
    Oidc.Log.ERROR,
    Oidc.Log.WARN,
    Oidc.Log.INFO,
    Oidc.Log.DEBUG,
    5
  ])
}

OpenId.defaultProps = {
  level: 5
}

export const useOpenId = () => useContext(OpenIdContext)
