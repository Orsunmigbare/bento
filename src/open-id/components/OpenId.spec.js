import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import { OpenId } from "./OpenId"

describe("CallbackLogin", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <OpenId />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
