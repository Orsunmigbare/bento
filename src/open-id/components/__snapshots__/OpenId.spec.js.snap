// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallbackLogin Snapshots No props 1`] = `null`;
